File: solution/services/book-service/app/env.py
import logging
import os
from dotenv import load_dotenv
from pathlib import Path

logging.basicConfig(
    level=logging.INFO,
    format="%(asctime)s %(levelname)s:  %(message)s",
    handlers=[logging.StreamHandler()]
)

env_path = Path(__file__).resolve().parent / ".env"
load_dotenv(dotenv_path=env_path)

logging.info("environment variables:")
logging.info("DATABASE_URL: %s", os.getenv("DATABASE_URL"))
logging.info("DATABASE_URL_MIGRATION: %s", os.getenv("DATABASE_URL_MIGRATION"))
logging.info("JWKS_URL: %s", os.getenv("JWKS_URL"))
logging.info("JWT_AUDIENCE: %s", os.getenv("JWT_AUDIENCE"))
logging.info("RABBITMQ_HOST: %s", os.getenv("RABBITMQ_HOST"))
logging.info("RABBITMQ_BOOKS_EXCHANGE: %s", os.getenv("RABBITMQ_BOOKS_EXCHANGE"))
logging.info("----------------------")

File: solution/services/user-service/app/env.js
import dotenv from 'dotenv';
import path from 'path'
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
const env_path = path.resolve(__dirname, '.env');

dotenv.config({path: env_path});

export const DATABASE_URL = process.env.DATABASE_URL;
export const JWKS_URL = process.env.JWKS_URL;
export const JWT_AUDIENCE = process.env.JWT_AUDIENCE;
export const RABBITMQ_HOST = process.env.RABBITMQ_HOST;
export const RABBITMQ_HEARTBEAT_EXCHANGE = process.env.RABBITMQ_HEARTBEAT_EXCHANGE;

console.log("environment variables:");
console.log("DATABASE_URL:", DATABASE_URL);
console.log("JWKS_URL:", JWKS_URL);
console.log("JWT_AUDIENCE:", JWT_AUDIENCE);
console.log("RABBITMQ_HOST:", RABBITMQ_HOST);
console.log("RABBITMQ_HEARTBEAT_EXCHANGE:", RABBITMQ_HEARTBEAT_EXCHANGE);
console.log("----------------------");
